samplecode/singularity/kw_salutations.py

```diff
@@ -0,0 +1,41 @@
+"""Given a file of names and a file of salutations to give each
+name, make an output file with greetings for each name in the input file.
+
+This script demonstrates the use of single-valued keyword-style parameters
+to Kive pipelines.
+"""
+
+import argparse
+import csv
+import typing as ty
+
+PARSER = argparse.ArgumentParser()
+PARSER.add_argument("--names", type=argparse.FileType())
+PARSER.add_argument("--salutations", type=argparse.FileType())
+PARSER.add_argument("outputfile", type=argparse.FileType("w"))
+
+
+def greet(name: str, salutation: str = "Hello") -> str:
+    return f"{salutation} {name}!"
+
+
+def get_salutations(rdr: csv.DictReader) -> ty.Dict[str, str]:
+    return {row["name"]: row["salutation"] for row in rdr}
+
+
+def main() -> None:
+    args = PARSER.parse_args()
+
+    names_reader = csv.DictReader(args.names)
```

Comment:
Keyword arguments are optional, so the example code should demonstrate checking for `None` and using some default value.

Reply:
Updated.

I made one argument mandatory and added `None` handling to the other, to demonstrate all the possibilities. I know required `--keyword-style` arguments are discouraged, but I think they can be valuable to just give names to required arguments (e.g. is it `ln source dest` or `ln dest source`?).

Reply:
That said, since keyword-style args are always optional as far as Kive is concerned, I guess this does introduce the possibility of a-priori invalid runs. If you think it best to always treat keyword-style parameters as optional I can update the example accordingly.

Reply:
That's what I was thinking: Kive doesn't support required keyword arguments, so let's keep the example simple. Always treat keyword arguments as optional.